src/test/regress/sql/multi_upgrade_reference_table.sql

```diff
@@ -423,9 +423,6 @@ FROM
 WHERE
     logicalrelid = 'upgrade_reference_table_transaction_rollback'::regclass;
 
--- eliminate the duplicate intermediate duplicate rows in pg_dist_colocation
```

Comment:
> We really shouldn't ever add code like this.

Yes, that makes sense. Because re-producing it was quite hard and seemingly not broke any distributed queries, to unblock things we checked-in first. Later, opened the issue immediately and started working on. We wouldn't have done so if that happened on a slightly more common usage pattern.